services/src/main/java/org/keycloak/protocol/ciba/decoupledauthn/HttpAuthenticationChannelProvider.java

```diff
@@ -0,0 +1,296 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.protocol.ciba.decoupledauthn;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.jboss.logging.Logger;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.broker.provider.util.SimpleHttp;
+import org.keycloak.common.util.Time;
+import org.keycloak.events.Details;
+import org.keycloak.events.Errors;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.ClientScopeModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.UserModel;
+import org.keycloak.models.utils.KeycloakModelUtils;
+import org.keycloak.protocol.ciba.CIBAAuthReqId;
+import org.keycloak.protocol.ciba.CIBAConstants;
+import org.keycloak.protocol.ciba.endpoints.request.BackchannelAuthenticationRequest;
+import org.keycloak.protocol.ciba.resolvers.CIBALoginUserResolver;
+import org.keycloak.protocol.ciba.utils.CIBAAuthReqIdParser;
+import org.keycloak.protocol.ciba.utils.DecoupledAuthStatus;
+import org.keycloak.protocol.ciba.utils.DecoupledAuthnResult;
+import org.keycloak.protocol.ciba.utils.DecoupledAuthnResultParser;
+import org.keycloak.services.CorsErrorResponseException;
+import org.keycloak.services.Urls;
+
+public class HttpAuthenticationChannelProvider extends HttpAuthenticationChannelProviderBase {
+
+    private static final Logger logger = Logger.getLogger(HttpAuthenticationChannelProvider.class);
+
+    protected final String httpAuthenticationRequestUri;
+
+    public HttpAuthenticationChannelProvider(KeycloakSession session, String httpAuthenticationRequestUri) {
+        super(session);
+        this.httpAuthenticationRequestUri = httpAuthenticationRequestUri;
+    }
+
+    protected String scope;
+    protected String userSessionIdWillBeCreated;
+    protected String userIdToBeAuthenticated;
+    protected String authResultId;
+    protected int expiration;
+
+    @Override
+    protected String getScope() {
+        return scope;
+    }
+
+    @Override
+    protected String getUserSessionIdWillBeCreated() {
+        return userSessionIdWillBeCreated;
+    }
+
+    @Override
+    protected String getUserIdToBeAuthenticated() {
+        return userIdToBeAuthenticated;
+    }
+
+    @Override
+    protected String getAuthResultId() {
+        return authResultId;
+    }
+
+    @Override
+    protected int getExpiration() {
+        return expiration;
+    }
+
+    @Override
+    protected Response verifyDecoupledAuthnResult() {
```

Comment:
Does it make more sense to move all this method to the `CIBADecoupledAuthenticator` ?

Please, see my last comment about whether we need a CIBA flow.

Reply:
@pedroigor I see, I've renamed this method.